### PR TITLE
TASK: Add support for horizontal scrolling in EditModePanel

### PR DESCRIPTION
**Pull request by dfeyer**

### CODE REVIEW

**Comment by dfeyer** [comment]
The framerate of the gif is not good at all ;)

I also work on this panel to implement the horizontal scrolling of the button

**Comment by dfeyer** [comment]
A bit better then what we have now


Need more love, but first let's merge this one.

**Comment by skurfuerst** [comment]
I love it :)

File: packages/neos-ui/src/Containers/EditModePanel/style.css [hunk 1]
```diff
@@ -29,12 +28,17 @@
...
+
+.editModePanel__previewModes,
+.editModePanel__editingModes {
+    width: 50%;
+    margin-right: 2rem;
 }
-.editModePanel--button {
+
+.editModePanel__button {
     margin-right: 1rem;
```
Comment (samuelhauser) [inline packages/neos-ui/src/Containers/EditModePanel/style.css]: maybe set margin-right to 0 on the last element?

**Comment by dfeyer** [comment]
> very nice! but the scrolling is a little bit strange. and some notification could be helpful

Yes lots of opportunity here to improve the UI I'm working on it, but basically we can merge this one if others agree and improve in an other PR

**Comment by dfeyer** [comment]
If your are OK i want to add the currently selected mode visible in the title, because of the scroll the current active one can be invisible.

File: packages/neos-ui/src/Containers/EditModePanel/style.css (hunk 1 above)
Comment (dfeyer) [inline packages/neos-ui/src/Containers/EditModePanel/style.css]: Fixed in the last version

**Comment by dfeyer** [comment]
> If your are OK i want to add the currently selected mode visible in the title, because of the scroll the current active one can be invisible.

Include in the last commit

@skurfuerst I refactor the component a bit, can you check, and if I have something to learn, let me know.

**Comment by dfeyer** [comment]
A little bit missing, highlight of the current editing mode in the toggler

**Pull request merged**